src/QGCApplication.cc

```diff
@@ -503,31 +503,39 @@ void QGCApplication::criticalMessageBoxOnMainThread(const QString& title, const
 #endif
 }
 
-#ifndef __mobile__
 void QGCApplication::saveTelemetryLogOnMainThread(QString tempLogfile)
 {
-    // The vehicle is gone now and we are shutting down so we need to use a message box for errors to hold shutdown and show the error
-    if (_checkTelemetrySavePath(true /* useMessageBox */)) {
-
-        QString saveDirPath = _toolbox->settingsManager()->appSettings()->telemetrySavePath();
-        QDir saveDir(saveDirPath);
-
-        QString nameFormat("%1%2.tlog");
-        QString dtFormat("yyyy-MM-dd hh-mm-ss");
-
-        int tryIndex = 1;
-        QString saveFileName = nameFormat.arg(QDateTime::currentDateTime().toString(dtFormat)).arg("");
-        while (saveDir.exists(saveFileName)) {
-            saveFileName = nameFormat.arg(QDateTime::currentDateTime().toString(dtFormat)).arg(QStringLiteral(".%1").arg(tryIndex++));
-        }
-        QString saveFilePath = saveDir.absoluteFilePath(saveFileName);
+    // Are we supposed to log?
+    if(toolbox()->settingsManager()->appSettings()->telemetrySave()->rawValue().toBool()) {
```

Comment:
The only case would be if you crash and there is a leftover log file. But if that happens then save was on at the time you crashed. Hence you would want to save that anyway.